### DISCUSSION TITLE
Sequoia 15.1 upgrade trashed my homebrew mariadb install and overwrote it with MySQL Community server

### DISCUSSION THREAD

**Discussion by vfontjr** [Homebrew/discussions#5698] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: 91f3671b16fe62121364c465c14e0dfd088d7ead
Last commit: 2 days ago
Core tap JSON: 29 Oct 00:06 UTC
Core cask tap JSON: 29 Oct 00:06 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 3.3.5 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: 16-core 64-bit arm_palma
Clang: 16.0.0 build 1600
Git: 2.47.0 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1-arm64
CLT: 16.1.0.0.1.1729049160
Xcode: 16.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libcmuclmtk.0.dylib
  /usr/local/lib/libcmuclmtk.dylib
  /usr/local/lib/libpocketsphinx.3.dylib
  /usr/local/lib/libpocketsphinx.dylib
  /usr/local/lib/libpocketsphinx_jni.dylib
  /usr/local/lib/libsphinxad.3.dylib
  /usr/local/lib/libsphinxad.dylib
  /usr/local/lib/libsphinxbase.3.dylib
  /usr/local/lib/libsphinxbase.dylib

Warning: Unbrewed '.la' files were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected '.la' files:
  /usr/local/lib/libcmuclmtk.la
  /usr/local/lib/libpocketsphinx.la
  /usr/local/lib/libsphinxad.la
  /usr/local/lib/libsphinxbase.la

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libcmuclmtk.a
  /usr/local/lib/libpocketsphinx.a
  /usr/local/lib/libsphinxad.a
  /usr/local/lib/libsphinxbase.a
```

### Description of issue

I'm a software developer and run mariadb installed through homebrew. When I first upgraded to Sequoia, my mariadb installation got trashed. I reinstalled and when the next bugfix release came out, I stopped all brew services to run the OS update and everything turned out fine. Mariadb was no longer installed.

I did the same this time with the 15.1 upgrade, but when I went to restart the homebrew services, mariadb wouldn't start. The brew services command showed mariadb's status as none.

Nevertheless, I opened PHPMyAdmin on local host and was shocked that it came up without error even with mariadb uninstalled. Then I saw this:

"Server version: 8.0.12 - MySQL Community Server - GPL"

There is also a startup item in the system settings for Oracle. I did not install this software and find it highly intrusive that Apple updates break homebrew installations.

**Comment by p-linnane** [comment]
> I did not install this software and find it highly intrusive that Apple updates break homebrew installations.

It seems like you're encountering multiple issues and conflating them. Things installed via `brew` are not removed by macOS updates, so we need to dig in deeper here. What is the output of `brew list`?

**Reply by vfontjr** [reply]
I wish I were conflating things. This is a MacBook MB3-ProMax completely maxed out. When I set it up as my new dev environment, I installed mariadb. I have mariadb installed on my dedicated server and I wanted to make sure my development server matched. Mariadb started going crazy when I upgraded to Sequoia. This is the output from brew services:

Name    Status  User       File
httpd   started victorfont ~/Library/LaunchAgents/homebrew.mxcl.httpd.plist
mariadb none               
php     started victorfont ~/Library/LaunchAgents/homebrew.mxcl.php.plist

Last night, I reinstalled mariadb, turned off MySql, and checked the database with mariadb running. The entire database that had existed before the upgrade had been transferred to MySql.

Also, when I stopped the mariadb brew service and restarted MySQL, brew services showed mariadb with the correct stopped status. Today, mariadb is gone again and cannot be restarted because it doesn't exist.

This is so befuddling, I asked ChatGPT what could have possibly happened. Here are a few plausible explanations for how MySQL might have replaced MariaDB:

1.  macOS Update Packages: While macOS doesn’t typically include MySQL by default, some updates could make changes to the Homebrew environment or reinstall packages to ensure compatibility, especially if it detects dependencies on core database services. It’s possible that the Sequoia 15.1 upgrade prompted Homebrew to install MySQL to satisfy some dependency or maintain the environment's stability.

2. Homebrew Self-Updates: Homebrew often self-updates when you run it after a system update. It may have switched MariaDB to MySQL if there was a formula update or if Homebrew’s dependencies shifted, which sometimes happens when certain packages are deprecated or considered non-standard. In this case, stopping MariaDB before the upgrade may not have prevented Homebrew from making changes to dependencies during or after the OS update.

3. Database Folder Conflicts: If MariaDB and MySQL shared or pointed to the same directories, the macOS update could have caused a reset or reinstallation of database services, installing MySQL to replace the MariaDB binaries while preserving database files.

4. Automated System Changes: Occasionally, system configuration files may reset after major macOS updates. If any part of the system or a package manager like Homebrew detected a need for MySQL as part of a compatibility fix, it could have triggered an automatic installation.

The final explanation is this freakin' Apple Silicon chip is haunted.

**Reply by SMillerDev** [reply]
I can confidently say that all four of the ChatGPT guesses are wrong. Especially the Homebrew one since it talks about things Homebrew can't know. 

If homebrew is still installed after a reboot, but one package is no longer installed I'd recommend looking if the problem also occurs if you log out. If it does I'm fairly certain you have the uninstall command, or brew bundle cleanup somewhere in your shell init scripts.

**Comment by vfontjr** [comment]
How can I look if the problem occurs when I log out. I need access to terminal. I have to be logged in for that?

If there i an uninstall script or brew cleanup in the shell init script, how did id get there. I don't do anything with shell scripts. I don't know how to.

**Reply by SMillerDev** [reply]
When you log out, your MariaDB install will have disappeared after logging in again, if the problem is what I expect it to be.
